### PR TITLE
Checkout item with patrons search

### LINKED ISSUE #227: ui: remove hardcoded user pids
- [ ] refactor the loan creation from an item to be able to select a patron (currently it is hardcoded)

### PR DESCRIPTION
**Pull request by vlad-bm**
Adds the ability to checkout an item from the items page and search and assign a patron to it.

(Closes #227 #239)

### CODE REVIEW

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/ItemDetails/components/ItemMetadata/components/CreateNewLoanModal/PatronsSearch/components/ResultsList/ResultsList.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { formatter } from '../../../../../../../../../../common/components/ResultsTable/formatters';
+import { pick } from 'lodash/object';
+
+export class ResultsList extends Component {
+  constructor(props) {
+    super(props);
+    this.viewDetailsClickHandler = this.props.viewDetailsClickHandler;
+  }
+
+  _rowActions = row => {};
```
Comment (equadon) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/ItemDetails/components/ItemMetadata/components/CreateNewLoanModal/PatronsSearch/components/ResultsList/ResultsList.js]: Is this something we need?

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/ItemDetails/components/ItemMetadata/components/CreateNewLoanModal/CreateNewLoanModal.js [hunk 2]
```diff
@@ -45,14 +55,122 @@ export default class CreateNewLoanModal extends Component {
...
+  renderSelection = () => {
+    return this.state.currentSelection ? (
+      <Label as="a" color="blue">
+        {this.state.currentSelection.patronEmail}
+        <Icon name="delete" onClick={this.handleRemoveSelection} />
+      </Label>
+    ) : null;
+  };
+
+  renderSuccess = () => {
```
[the remark this answers is not in the document]
Comment (vlad-bm) [reply invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/ItemDetails/components/ItemMetadata/components/CreateNewLoanModal/CreateNewLoanModal.js]: I wanted to use notifications, but the modal dims the background, and it should also remain open until the user either dismisses it or clicks to view the newly created loan details.

File: invenio_app_ils/ui/src/invenio_app_ils/common/components/PatronSearchBox/PatronSearchBox.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    super(props);
+    this.minCharacters = props.minCharacters;
+    this.state = {
+      isLoading: false,
+      value: '',
+      results: [],
+    };
+  }
+
+  mapResults = hits => {
```
[the remark this answers is not in the document]
Comment (vlad-bm) [reply invenio_app_ils/ui/src/invenio_app_ils/common/components/PatronSearchBox/PatronSearchBox.js]: We decided not to use serializeSearchResults anymore, so it was removed when set-access-restrictions was merged

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/ItemDetails/components/ItemMetadata/components/CreateNewLoanModal/CreateNewLoanModal.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kpsherva) [reply invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/ItemDetails/components/ItemMetadata/components/CreateNewLoanModal/CreateNewLoanModal.js]: I'm not sure if I understand about the background. I agree that here we should use the notifications - if the notification is not visible due to modal background, you can control it with css.

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/PatronDetails/components/ItemsCheckout/state/actions.js [hunk 4]
```diff
@@ -13,13 +13,18 @@ export const checkoutItem = (item, patronPid, shouldForceCheckout = false) => {
       type: IS_LOADING,
     });
     const currentUser = sessionManager.user;
+    const loan = {
+      metadata: {
+        item_pid: item.item_pid,
+        patron_pid: String(patronPid),
```
Comment (kpsherva) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/PatronDetails/components/ItemsCheckout/state/actions.js]: the patron pid should be serialized to string in the serializer (in /api/user.js), as it comes as a number but everywhere we use it as the string

**Review by zzacharo — DISMISSED** [review]

**Pull request merged**